### ISSUE TITLE
Gaussian Graphs are nice but statistically not correct

### ISSUE THREAD

**Issue by furstblumier** [IAmPythagoras/FFXIV-Combat-Simulator#13]
Hi. First of all, nice Project. I'm the creator of the BLM sheets, and a bunch of other sheets as well.

You say here in your code that https://github.com/IAmPythagoras/FFXIVPPSCalculator/blob/39d2d534f727257432ee6594fcc8fd50857f5b01/Fight.py#L105 
This is obviously true in the general sense. Binom limits in poisson which limits in a normal distribution. This is only true though for several million samples. From your documentation, n is the number of spells in a sequence, which is far far below that number.

I doubt I need to illustrate the issue but let's do it anyway
![grafik](https://user-images.githubusercontent.com/7149167/191110389-b8125755-6289-486b-9a73-ffc4166c685b.png)
This is the distribution of a long BLM rotation with a few thousand samples.
![grafik](https://user-images.githubusercontent.com/7149167/191110510-7c5c6aef-be57-4b29-82fd-5408b39a7784.png)
here for a few 100k samples
![grafik](https://user-images.githubusercontent.com/7149167/191110559-dbcd93d2-2b7e-475a-b284-a896de076513.png)
and for a few millions. We see the limit here for _millions of iterations of a long sequence of spells_.
Now for comparison the sequence `F4 F4 F4 F1 F4 F4 swift F4`:
![grafik](https://user-images.githubusercontent.com/7149167/191110772-81143179-f08e-4c2c-8e19-4a9be575b944.png)
As you can see here, even with millions of samples, this looks far from normal. (The Graphs were made by Nemekh several years ago, many thanks for that again).

Imo the approach of making distributions in this project is completely backwards: You assume a mean value, then assume a distribution and calculate sigma and thus graph the spread. Instead you should just monte carlo it. Array and calculate each individual damage event (not the mean! The real event, Factoring in crit chance and buffs etc) and make a histogram from that. Then Calculate a sigma based on this data. 

You might ask "why are you not doing that in your cheats then?" Because I can't. I don't calculate indivisual damage events. I calculate essentially a single hit with potency of my pps value. Without individual hits I need to do some tricks to make it work. You don't have this issue so it would be way better to just do it better.

**Comment by IAmPythagoras** [comment]
Hey! Yeah, I have been thinking some time ago to revisit those graphs I was generating. And as you said, I believe they were wrong. I decided not too spend too much time on them again because :

1- my knowledge of statistic is actually not very in depth. I had a probability theory class (that in the end went more toward measure theory :x ) and proved not that useful to what I wanted to do with this sim.

2- I wanted to make sure the core of the simulator was working rather than what I was doing with the outputs.

However, thanks a lot for your input, I am taking a statistic class this semester, so I guess in time I will be better at understanding all these things haha.

Iirc, my judgment for saying what I said in the graph was that I was that (as you said) the binomial distribution approaches a normal distribution as the number of sample goes to infinity. I remember checking how much sample we were talking about tho, and I found out that around n=9 the error was quite small. So at the time I decided that I had plenty enough n and that it would have been fine. Either the information was wrong or I missunderstood what I read, which in the end doesn't matter much.

I will look into fixing those graph soon. I think having a distribution graph is as important as having the average DPS for the simulator since it could tell you if a comp or build can result in a very high DPS if you are very lucky. Which might be of interest for people trying to speedrun raids.

In any case, thanks a lot again. I will look into this more in depth soon:)
